src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -448,23 +426,18 @@ private MsalAccessTokenCacheItem GetUnexpiredAccessTokenOrNull(AuthenticationReq
 
         private static MsalAccessTokenCacheItem GetSingleResult(AuthenticationRequestParameters requestParams, IEnumerable<MsalAccessTokenCacheItem> filteredItems)
         {
-            MsalAccessTokenCacheItem msalAccessTokenCacheItem;
-
-            // if only one cached token found
-            if (filteredItems.Count() == 1)
+            try
             {
-                msalAccessTokenCacheItem = filteredItems.First();
-            }
-            else
+                return filteredItems.Single();
+            } 
+            catch (InvalidOperationException)
```

Comment:
Let's not use exceptions for control flow. See https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why

Reply:
Done. Reverted back.